crypto.go

```diff
@@ -157,8 +163,16 @@ func (c *secretCache) initKeys() error {
 	return nil
 }
 
-func deriveMasterKey(password []byte, email string, iter int) []byte {
-	return pbkdf2.Key(password, []byte(strings.ToLower(email)), iter, 32, sha256.New)
+func deriveMasterKey(password []byte, email string, kdfType int, iter int, mem int, par int) []byte {
+	switch kdfType {
+	case KDFTypePBKDF2:
+		return pbkdf2.Key(password, []byte(strings.ToLower(email)), iter, 32, sha256.New)
+	case KDFTypeArgon2id:
+		var salt [32]byte = sha256.Sum256([]byte(strings.ToLower(email)))
+		return argon2.IDKey(password, salt[:], uint32(iter), uint32(mem*1024), uint8(par), 32)
+	default:
+		return pbkdf2.Key(password, []byte(strings.ToLower(email)), iter, 32, sha256.New)
```

Comment:
this duplicates KDFTypePBKDF2, so I'd make it an error as presumably it's a type we don't yet support.

Reply:
Thank you for the quick review!

I changed this in a47b294 so that we now return an error in case it is not one of the defined KDFTypes.

Also, I made the KDFType it's own type instead of being int, similar to how it is already done with CipherStringType.